examples/gdp/farm_layout/farm_layout.py

```diff
@@ -0,0 +1,274 @@
+        'width_lbs': {1: 1, 2: 1, 3: 1, 4: 1, 5: 1},
+        'length_upper_overall': 30,
+        'width_upper_overall': 30,
+    },
+    "FLay06": {
+        'areas': {1: 40, 2: 50, 3: 60, 4: 35, 5: 75, 6: 20},
+        'length_lbs': {1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1},
+        'width_lbs': {1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1},
+        'length_upper_overall': 30,
+        'width_upper_overall': 30,
+    },
+    # An example where the dimension constraints are strict enough that it cannot fit a square
+    "FLay03_alt_1": {
+        'areas': {1: 40, 2: 50, 3: 60},
+        'length_lbs': {1: 5, 2: 5, 3: 5},
+        'width_lbs': {1: 5, 2: 5, 3: 7},
+        'length_upper_overall': 30,
+        'width_upper_overall': 30,
+    },
+    "FLay03_alt_2": {
+        'areas': {1: 40, 2: 50, 3: 60},
+        'length_lbs': {1: 1, 2: 1, 3: 1},
+        'width_lbs': {1: 1, 2: 1, 3: 1},
+        'length_upper_overall': 8,
+        'width_upper_overall': 30,
+    },
+}
+
+
+def build_model(params=farm_layout_model_examples["FLay03"]):
+    """Build the model. Default example gives FLay03 which is fairly small"""
+
+    # Ensure good data was passed
+    assert (
+        len(params) == 5
+    ), "Params should be in the format: areas, length_lbs, width_lbs, length_upper_overall, width_upper_overall"
+
+    areas = params['areas']
+    length_lbs = params['length_lbs']
+    width_lbs = params['width_lbs']
+    length_upper_overall = params['length_upper_overall']
+    width_upper_overall = params['width_upper_overall']
+
+    for a in areas:
+        assert a > 0, "Area must be positive"
+
+    m = ConcreteModel(name="Farm Layout model")
+    m.plots = RangeSet(len(areas), doc=f"{len(areas)} plots")
+    m.plot_pairs = Set(initialize=m.plots * m.plots, filter=lambda _, p1, p2: p1 < p2)
+
+    m.length_lbs = Param(
+        m.plots,
+        initialize=length_lbs,
+        doc="Lower bounds for length of individual rectangles",
+    )
+    m.width_lbs = Param(
+        m.plots,
+        initialize=width_lbs,
+        doc="Lower bounds for length of individual rectangles",
+    )
+    m.plot_area = Param(m.plots, initialize=areas, doc="Area of this plot")
+
+    m.overall_length_ub = Param(
+        initialize=length_upper_overall, doc="Overall upper bound for length"
+    )
+    m.overall_width_ub = Param(
+        initialize=width_upper_overall, doc="Overall upper bound for width"
+    )
+
+    m.Length = Var(
+        domain=NonNegativeReals, doc="Overall length of the farmland configuration"
+    )
+    m.Width = Var(
+        domain=NonNegativeReals, doc="Overall width of the farmland configuration"
+    )
+
+    m.perim = Objective(
+        expr=2 * (m.Length + m.Width), doc="Perimeter of the farmland configuration"
+    )
+
+    m.plot_x = Var(
+        m.plots, bounds=(0, m.overall_length_ub), doc="x-coordinate of plot center"
+    )
+    m.plot_y = Var(
+        m.plots, bounds=(0, m.overall_width_ub), doc="y-coordinate of plot center"
+    )
```

Comment:
Are these `doc` arguments right? I thought x and y were bottom lefthand corner in this implementation?

Reply:
Oops, you're right. I had to change that at some point.